lib/editor/components/timetable/TripSeriesModal.js

```diff
@@ -114,42 +127,59 @@ const TripSeriesModal = (props: Props) => {
+            <HourMinuteInput
+              onChange={(value) => setState((prevState) => ({ ...prevState, startTime: value }))}
+              seconds={startTime}
+              showSeconds={useSecondsInOffset}
+              standaloneInput
+            />
           </div>
           <div style={{marginLeft: '20px'}}>
             {messages('headway')}
-            <HourMinuteInput onChange={setHeadway} seconds={headway} showSeconds={useSecondsInOffset} standaloneInput />
+            <HourMinuteInput
+              onChange={(value) => setState((prevState) => ({ ...prevState, headway: value }))}
+              seconds={headway}
+              showSeconds={useSecondsInOffset}
+              standaloneInput
+            />
           </div>
           <div style={{margin: '0px 0px 0px 20px'}}>
             {messages('endTime')}
-            <HourMinuteInput onChange={setEndTime} seconds={endTime} showSeconds={useSecondsInOffset} standaloneInput />
+            <HourMinuteInput
+              onChange={(value) => setState((prevState) => ({ ...prevState, endTime: value }))}
+              seconds={endTime}
+              showSeconds={useSecondsInOffset}
+              standaloneInput
+            />
           </div>
         </div>
         <hr />
         <div>
-          <Checkbox checked={useAutoTripIds} onChange={handleAutoTripIDCheckBox}>{messages('automaticallyUpdateTripIds')}</Checkbox>
-          {useAutoTripIds &&
+          <Checkbox checked={useAutoTripIds} onChange={() => setState((prevState) => ({ ...prevState, useAutoTripIds: !prevState.useAutoTripIds }))}>
+            {messages('automaticallyUpdateTripIds')}
+          </Checkbox>
+          {useAutoTripIds && (
             <>
               <div style={{alignItems: 'center', display: 'flex'}}>
                 <>
                   <FormControl
-                    onChange={handleTripPrefixUpdate}
+                    onChange={(evt) => setState((prevState) => ({ ...prevState, autoTripIdPrefix: evt.target.value }))}
                     placeholder={messages('prefixPlaceholder')}
                     style={{width: '30%', marginRight: '5px'}}
                     type='text'
                     value={autoTripIdPrefix}
                   />
-                -
+                  -
                   <FormControl
-                    onChange={handleIncrementStartUpdate}
+                    onChange={(evt) => setState((prevState) => ({ ...prevState, autoTripIdStart: +evt.target.value }))}
                     placeholder={messages('incrementStartPlaceholder')}
                     style={{width: '40%', marginLeft: '5px', marginRight: '5px'}}
                     type='number'
                   />
                   {messages('incrementBy')}
                   <FormControl
-                    onChange={handleTripIncrementUpdate}
+                    onChange={(evt) => setState((prevState) => ({ ...prevState, autoTripIdIncrement: +evt.target.value }))}
```

Comment:
I think Phil told you to get rid of this method but I'm saying bring it back. this line is complex enough we'll want to keep it in a function

Reply:
Agree on this one

Reply:
fair